apps/fyler/src/fyler_category_manager.erl

```diff
@@ -0,0 +1,694 @@
+  {NextState, NewState} = choose_pool(State1),
+  {next_state, NextState, NewState};
+
+pools_({task_rejected, Ref, Node}, #state{category = Category, retasks = ReTasks, pools = Pools} = State) ->
+  case status(Node, Pools) of
+    {pending, Ref, Id} ->
+      NewPools = enable(Node, Pools),
+      case task(Category, Id) of
+        undefined ->
+          {next_state, pools, State#state{pools = NewPools}};
+        Task ->
+          {NextState, NewState} = choose_pool(State#state{retasks = fyler_queue:in(Task, ReTasks), pools = NewPools}),
+          {next_state, NextState, NewState}
+      end;
+    _ ->
+      {next_state, pools, State}
+  end;
+
+pools_({run, #task{priority = Priority} = Task}, #state{tasks = Tasks, priorities = Priorities} = State) ->
+  update_ets(Task),
+  NumPriority = maps:get(Priority, Priorities, 1),
+  NewTasks = fyler_queue:in(Task, NumPriority, Tasks),
+  {NextState, NewState} = choose_pool(State#state{tasks = NewTasks}),
+  {next_state, NextState, NewState};
+
+pools_({cancel_task, Id}, #state{category = Category} = State) ->
+  case ets:lookup(Category, Id) of
+    [#current_task{pool = Node, status = progress}] ->
+      ets:delete(Category, Id),
+      rpc:cast(Node, fyler_pool, cancel_task, [Id]);
+    [#current_task{status = queued}] ->
+      ets:delete(Category, Id);
+    _ ->
+      ok
+  end,
+  {next_state, pools, State}.
+
+tasks_(start, State) ->
+  {next_state, tasks, send_check(State)};
+
+tasks_(check, #state{passive_pools = []} = State) ->
+  {next_state, tasks, send_check(State)};
+
+tasks_(check, #state{passive_pools = Pools, node_to_id = NodeToId} = State) ->
+  start_pool(lists:last(Pools), NodeToId),
+  {next_state, tasks, send_check(State)};
+
+tasks_({pool_connected, Node}, #state{pools = Pools, passive_pools = PassivePools} = State) ->
+  State1 = State#state{pools = connect(Node, Pools), passive_pools = lists:delete(Node, PassivePools)},
+  {NextState, NewState} = choose_pool(State1),
+  {next_state, NextState, NewState};
+
+tasks_({pool_enabled, Node}, #state{pools = Pools} = State) ->
+  {NextState, NewState} = choose_pool(State#state{pools = enable(Node, Pools)}),
+  {next_state, NextState, NewState};
+
+tasks_({pool_down, Node}, #state{category = Category, retasks = ReTasks, pools = Pools, passive_pools = []} = State) ->
+  NewReTasks = restart_tasks(Category, ReTasks, Node),
+  {next_state, tasks, State#state{retasks = NewReTasks, pools = delete(Node, Pools), passive_pools = [Node]}};
+
+tasks_({pool_down, Node}, #state{category = Category, retasks = ReTasks, pools = Pools,
+  passive_pools = PassivePools} = State) ->
+
+  NewReTasks = restart_tasks(Category, ReTasks, Node),
+  NewState = State#state{retasks = NewReTasks, pools = delete(Node, Pools), passive_pools = [Node|PassivePools]},
+  {next_state, tasks, NewState};
+
+tasks_({pool_disabled, Node}, #state{category = Category, retasks = ReTasks, pools = Pools} = State) ->
```

Comment:
Line 381 is too long: tasks_({pool_disabled, Node}, #state{category = Category, retasks = ReTasks, pools = Pools} = State) ->.